### PR TITLE
Comvert recipe subdirectory to typescript

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: config/tslint.base.json [hunk 1]
```diff
@@ -23,10 +23,8 @@
     // mark everything as public. Talk to shans@ if you disagree.
     // "member-access": [true],
     "new-parens": true,
-    "no-angle-bracket-type-assertion": true,
```
[the remark this answers is not in the document]
Comment (lindner) [reply config/tslint.base.json]: ack. agree with rationale too

File: config/tslint.base.json [hunk 2]
```diff
@@ -23,10 +23,8 @@
...
     "no-any": true,
     "no-arg": true,
-    "no-conditional-assignment": true,
```
[the remark this answers is not in the document]
Comment (lindner) [reply config/tslint.base.json]: ack.

File: runtime/ts/recipe/connection-constraint.ts [hunk 3]
```diff
@@ -61,46 +70,54 @@ export class InstanceEndPoint {
 }
 
 export class HandleEndPoint {
+  handle: Handle;
```
Comment (lindner) [inline runtime/ts/recipe/connection-constraint.ts]: can be readonly

File: runtime/ts/recipe/connection-constraint.ts [hunk 4]
```diff
@@ -61,46 +70,54 @@ export class InstanceEndPoint {
...
   }
 
-  toString() {
+  toString(nameMap = undefined) {
     return `${this.handle.localName}`;
   }
 }
 
 export class TagEndPoint {
+  tags: string[];
```
Comment (lindner) [inline runtime/ts/recipe/connection-constraint.ts]: can be readonly

File: runtime/ts/recipe/util.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2017 Google Inc. All rights reserved.
```
Comment (lindner) [inline runtime/ts/recipe/util.ts]: github lost the rename here?  Thinks this is a new file...

File: runtime/ts/recipe/walker-base.ts [hunk 6]
```diff
@@ -33,27 +33,34 @@ import {assert} from '../../platform/assert-web.js';
  * if 3 methods return [a,b], [c,d,e], and [f] respectively,
  * then "Independent" will cause 6 outputs to be generated: [a,b,c,d,e,f]
  */
-export class WalkerBase extends Strategizer.Walker {
+
+export enum WalkerTactic {Permuted='permuted', Independent='independent'}
```
Comment (lindner) [inline runtime/ts/recipe/walker-base.ts]: nice

**Comment by shans** [comment]
With the rename of util.js, I _think_ what's happened is: git doesn't actually track file moves (even if you use git mv). So util.ts is small enough and changed enough for git to not auto-detect it as the "same" file.

I don't know if there is anything that I can do to fix that?

**Comment by lindner** [comment]
I think I can coerce it to show the mv if I use -M flag.

I wouldn't block on this.  Just an oddity that I noticed.

**Pull request merged**